listenbrainz/db/dump.py

```diff
@@ -897,25 +897,34 @@ def _update_sequences():
     _update_sequence(timescale.engine, 'playlist.playlist_recording_id_seq', 'playlist.playlist_recording')
 
 
-def _fetch_latest_file_info_from_ftp_dir(server, dir):
+def _fetch_latest_file_info_from_ftp_dir(directory: str, has_id: bool):
     """
-        Given a FTP server and dir, fetch the latest dump directory name and return it
+        Given a base FTP dir and whether the dump name contains an id, browses the MB FTP server to fetch
+        the latest dump directory name and return it
     """
 
-    line = ""
+    latest_dump_id: Optional[int] = None
+    latest_dt: Optional[datetime] = None
 
-    def add_line(l):
-        nonlocal line
-        l = l.strip()
-        if l:
-            line = l
+    def process_line(file):
+        nonlocal latest_dump_id, latest_dt
+        file = file[56:].strip()
```

Comment:
I checked and found that lines are like
```
drwxr-xr-x    2 ftp      ftp          4096 Aug 21 00:01 listenbrainz-dump-988-20220821-000003-incremental
```.

If we use NLST instead of LIST, then we get the file name only. (https://stackoverflow.com/questions/47830336/ftp-list-file-name-via-bash)